lib/commanded/commands/dispatcher.ex

```diff
@@ -50,6 +51,18 @@ defmodule Commanded.Commands.Dispatcher do
 
     pipeline = before_dispatch(pipeline, payload)
 
+    uuid_generator = Application.uuid_generator(pipeline.application, true)
+
+    uuid_updater = fn
+      nil -> uuid_generator.()
+      value -> value
+    end
+
+    pipeline =
+      pipeline
+      |> Map.update!(:command_uuid, uuid_updater)
```

Comment:
here you could have use [Map.put_new_lazy](https://hexdocs.pm/elixir/1.12/Map.html#put_new_lazy/3)

Reply:
Using `Map.put_new_lazy` does not work on structs since the key is always present and will be a NOOP.

Reply:
Oh you right I forgot about the struct!